src/Microsoft.Health.Fhir.Core/Models/SearchParameterInfo.cs

```diff
@@ -94,6 +94,14 @@ public SearchParameterInfo(string name)
         /// </summary>
         public bool IsPartiallySupported { get; set; }
 
+        /// <summary>
+        /// The component definitions if this is a composite search parameter (<see cref="Type"/> is <see cref="SearchParamType.Composite"/>)
+        /// </summary>
         public IReadOnlyList<SearchParameterComponentInfo> Component { get; }
+
+        /// <summary>
+        /// The resolved <see cref="SearchParameterInfo"/>s for each component if this is a composite search parameter (<see cref="Type"/> is <see cref="SearchParamType.Composite"/>)
+        /// </summary>
+        public IReadOnlyList<SearchParameterInfo> ResolvedComponents { get; set; } = Array.Empty<SearchParameterInfo>();
```

Comment:
>ResolvedComponents [](start = 50, length = 18)

nit: I would consider to move `SearchParameterInfo.Type` and `SearchParameterInfo.TargetResourceTypes` into `SearchParameterComponentInfo` to unify them. Having two, somewhat similar entities, can be a bit confusing (at least for me)

Reply:
Thank you for the suggestion. I've added a `ResolvedSearchParameter` property to `SearchParameterComponentInfo` and removed `ResolvedComponents` from `SearchParameterInfo`